pkg/sdk/tag_association_validations.go

```diff
@@ -3,21 +3,33 @@ package sdk
 var (
 	// based on https://docs.snowflake.com/en/user-guide/object-tagging.html#supported-objects
 	TagAssociationAllowedObjectTypes = []ObjectType{
+		// organization level
 		ObjectTypeAccount,
+
+		// account level
 		ObjectTypeApplication,
 		ObjectTypeApplicationPackage,
 		ObjectTypeDatabase,
+		ObjectTypeFailoverGroup,
 		ObjectTypeIntegration,
 		ObjectTypeNetworkPolicy,
+		ObjectTypeReplicationGroup,
 		ObjectTypeRole,
 		ObjectTypeShare,
 		ObjectTypeUser,
 		ObjectTypeWarehouse,
+
+		// database level
 		ObjectTypeDatabaseRole,
 		ObjectTypeSchema,
+
+		// schema level
 		ObjectTypeAlert,
+		ObjectTypeBudget,
+		ObjectTypeClassification,
 		ObjectTypeExternalFunction,
```

Comment:
We could create a test that would go through all the object types to see which ones are valid ones, I just checked that, e.g. dynamic tables can also be used (with table data type ofc).

Reply:
+1 for such a test

Reply:
It's related to https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3126#discussion_r1799149375 - I can do this in the next PR.